internal/tracing/tracer.go

```diff
@@ -19,12 +19,14 @@ import (
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
 	semconv "go.opentelemetry.io/otel/semconv/v1.18.0"
 	"go.opentelemetry.io/otel/trace"
+	"go.opentelemetry.io/otel/trace/embedded"
 
 	"github.com/canonical/identity-platform-admin-ui/internal/logging"
 )
 
 type Tracer struct {
 	tracer trace.Tracer
+	embedded.Tracer
```

Comment:
mind sending a link to the change in the lib? The line above this seems redundant unless the two tracer interfaces are completely dislinked

Reply:
Here's the [link](https://pkg.go.dev/go.opentelemetry.io/otel/trace#hdr-API_Implementations) to the paragraph.

TL;DR: this is due to the fact that the Otel library may present changes in the API even without a major version bump, so they use this golang "trick" (that I randomly learned last week from another library) to force you to notice the change with a compilation error.
The alternatives would be to not be alerted at all, or to be alerted with runtime panics.

So I chose this one to get alerted at compilation time. If you remove the embedeed struct we get compilation error because our object would not respect the interface.